src/plugins/discover/common/esql_locator.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DISCOVER_ESQL_LOCATOR } from '@kbn/deeplinks-analytics';
+import { LocatorDefinition, LocatorPublic } from '@kbn/share-plugin/common';
+import { SerializableRecord } from '@kbn/utility-types';
+
+export type DiscoverEsqlLocatorParams = SerializableRecord;
+
+export interface DiscoverEsqlLocatorDependencies {
+  discoverAppLocator: LocatorPublic<SerializableRecord>;
+  getIndices: (options: {
+    showAllIndices: boolean;
+    pattern: string;
+    isRollupIndex: () => boolean;
+  }) => Promise<Array<{ name: string }>>;
+}
+
+export type DiscoverEsqlLocator = LocatorPublic<DiscoverEsqlLocatorParams>;
+
+export class DiscoverEsqlLocatorDefinition implements LocatorDefinition<DiscoverEsqlLocatorParams> {
+  public readonly id = DISCOVER_ESQL_LOCATOR;
+
+  constructor(protected readonly deps: DiscoverEsqlLocatorDependencies) {}
+
+  public readonly getLocation = async () => {
```

Comment:
Is this correct, @stratoula ?

Reply:
Without having tested it you will need to also pass a dataviewSpec. So you need to do this:

1. After you have defined the indexName create an adhooc dataview
2. Create an adhoc like:
```
const dataView = await deps.dataViews.create(
       {
        title: indexName,
      }
    );
```
3. Pass the dataviewSpec in the params:

```
const params = {
          query: {
            esql: `from ${indexName} | limit 10`;
          },
          dataViewSpec: dataView?.toSpec(),
        };
```

Reply:
@stratoula Given your knowledge of the space, could you contribute an accurate commit to this PR?  Since we have folks from design (specifically @andreadelrio) looking at the visual treatment, we have some time to make this correct.

Reply:
@stratoula don't we also need to check if the preview is enabled?